### PR TITLE
Add missing description of numpy.concatenate behavior for axis=None.

### PR DESCRIPTION
**Pull request by carlosgmartin**
Adds a missing description for the behavior of [`jax.numpy.concatenate`](https://docs.jax.dev/en/latest/_autosummary/jax.numpy.concatenate.html) for `axis=None`.

### CODE REVIEW

**Review by jakevdp — APPROVED** [review]
Thanks!
